packages/calypso-e2e/src/lib/flows/gutenboarding-flow.ts

```diff
@@ -0,0 +1,205 @@
+		return url;
+	}
+
+	/**
+	 * Given a name, select a design matching the name.
+	 *
+	 * @param {string} name Name of the design to use.
+	 */
+	async selectDesign( name: string ): Promise< void > {
+		await Promise.all( [
+			this.page.waitForNavigation(),
+			this.page.click( selectors.design( name ) ),
+		] );
+	}
+
+	/**
+	 * Given one name of the font in a pair, select a matching font peairing.
+	 *
+	 * Example:
+	 * 	- name = Playfair will match [ Playfair / Fira Sans ]
+	 * 	- name = Roboto will match [ Space Mono / Roboto ]
+	 *
+	 * @param {string} name One portion of the name of the font pair.
+	 */
+	async selectFont( name: string ): Promise< void > {
+		// Font selector depends on the viewport name but lumps non-mobile into desktop.
+		const viewportName = getViewportName() === 'mobile' ? 'mobile' : 'desktop';
+
+		// Mobile viewport puts the buttons behind a dropdown.
+		if ( viewportName === 'mobile' ) {
+			await this.page.click( selectors.mobileFontPairingDropdown );
+		}
+
+		// Click on the desired font pairing.
+		await this.page.click( selectors.fontPairingButton( viewportName, name ) );
+
+		if ( viewportName === 'desktop' ) {
+			await this.page.waitForSelector( `button.is-selected span:text("${ name }")` );
+		} else {
+			await this.page.waitForSelector(
+				`button.is-selected-dropdown-option span:text("${ name }")`,
+				{ state: 'hidden' }
+			);
+		}
+	}
+
+	/**
+	 * Given an array of feature strings, selects matching features.
+	 *
+	 * @param {Features} features Array of features to select.
+	 */
+	async selectFeatures( features: Features[] ): Promise< void > {
+		for await ( const feature of features ) {
+			// Click on the feature to select, then make sure it is in fact selected by checking
+			// whether the attribute `is-selected` is present for the button with that feature.
+			await this.page.click( `${ selectors.featureItem } :text-is("${ feature }")` );
+			await this.page.waitForSelector(
+				`${ selectors.featureItem }.is-selected :text-is("${ feature }")`
+			);
+		}
+	}
+
+	/**
+	 * Given a name, select a plan matching the name.
+	 *
+	 * @param {string} name Name of the plan.
+	 */
+	async selectPlan( name: Plans ): Promise< void > {
+		// First, expand the accordion.
+		await this.page.click( ':text-is("Show all plans")' );
+
+		const plans = await this.page.$$( selectors.planItem );
+
+		// Iterate through the top-level container for each plan.
+		// Then, extract name of the plan and if it finds a match with the desired plan,
+		// click on the `Select` button within the container.
+		for await ( const plan of plans ) {
+			const planName = await plan
+				.waitForSelector( selectors.planName )
+				.then( ( el ) => el.innerText() );
+
+			console.log( planName );
+			if ( planName === name ) {
+				const button = await plan.waitForSelector( selectors.button( 'Select' ) );
+				await Promise.all( [ this.page.waitForNavigation(), button.click() ] );
+				return;
+			}
+		}
+	}
```

Comment:
I think we can do a much simpler implementation of this with just a slightly more fancy selector!

Here's an example of what I tried and worked, but feel free to iterate/clean-up :) 

```typescript
async selectPlan( name: Plans ): Promise< void > {
		// First, expand the accordion.
		await this.page.click( ':text-is("Show all plans")' );
		await Promise.all( [
			this.page.waitForNavigation(),
			this.page.click(
				`.plans-accordion-item:has(.plans-accordion-item__heading:has-text("${ name }")) button:has-text("Select")`
			),
		] );

		// const plans = await this.page.$$( selectors.planItem );

		// // Iterate through the top-level container for each plan.
		// // Then, extract name of the plan and if it finds a match with the desired plan,
		// // click on the `Select` button within the container.
		// for await ( const plan of plans ) {
		// 	const planName = await plan
		// 		.waitForSelector( selectors.planName )
		// 		.then( ( el ) => el.innerText() );

		// 	console.log( planName );
		// 	if ( planName === name ) {
		// 		const button = await plan.waitForSelector( selectors.button( 'Select' ) );
		// 		await Promise.all( [ this.page.waitForNavigation(), button.click() ] );
		// 		return;
		// 	}
		// }
	}
```

Reply:
Thanks for the suggestion, I'll give your selector a try.

Reply:
Interestingly, the suggested selector matches more than 1 item:

![image](https://user-images.githubusercontent.com/6549265/130537237-5f416313-87e5-480e-afa7-a2e8ce7d910c.png)

It is matching on the text that states `Pick a free domain (1 year)`.

It looks like that with the accordion expanded, it matches the 'Free' plan first, but I wonder if this would be the case if the order was shifted in the future.

A note, I struggled with the selector for the business plan for the longest time while working on this PR. I'm okay with accepting a selector that matches multiple items, but personally I think a traditional for/if combination is easier to follow.

Reply:
I've changed a portion of the selector to this and it seems to filter out the false match:
```
.plans-accordion-item:has(.plans-accordion-item__name:has-text("${name}")) button:has-text("Select")'
```

Reply:
Fixed in 32357f3ed0c917646b87a95c963f9109d720b478.

Reply:
> personally I think a traditional for/if combination is easier to follow

Interesting! That's helpful to hear, because I actually had the opposite gut reaction when I first read this! This might be worth discussing as a group later as there might be a style-guide point to make here. Complicated selection is inevitable, is it better to blow that out into DOM iterating, or condense into a single selector?

Reply:
I agree that it's worth having a discussion over. I prefer a for loop as Playwright makes it easy to select all matching top-level items. By iterating on the top-level items it's easy to limit search scope to the child elements.